src/errors.jl

```diff
@@ -471,6 +471,8 @@ function julia_error(
             ) #=error=#
             world = enzyme_extract_world(f)
         end
+        # TODO: get world from TypeAnalyzer
+        # @assert world == enzyme_gutils_context(gutils).world
         throw(IllegalTypeAnalysisException(msg, mi, world, sval, ir, bt))
```

Comment:
@wsmoses would you be okay with a change where `CustomErrorHandler` always receives `gutils`? We can go from `gutils` to `TypeAnalyzer` (if I saw that right).

Reply:
Depending on the particular error thrown there may not be a gutils created.

But we can for sure audit all the current error types and see what we can do